apps/05Mastermind.js

```diff
@@ -30,38 +28,42 @@ function generateSolution() {
 }
 
 function getRandomInt(min, max) {
-  return Math.floor(Math.random() * (max - min)) + min;
+    return Math.floor(Math.random() * (max - min)) + min;
 }
 
 function generateHint(solution, guess) {
-  var solutionArray = solution.split('');
-  var guessArray = guess.split('');
-  var correctLetterLocations = 0;
-  var correctLetters = 0;
-  for (var i = 0; i<= solutionArray; i++) {
-    if (solutionArray[i]===guessArray[i]) {
-      correctLetterLocations++;
-      solutionArray[i] = null;
-    }
-  }
-
-  for (var i = 0;i <= solutionArray; i++) {
-    var targetIndex = guessArray.indexOf(solutionArray[i]);
-    if (targetIndex > -1) {
-      correctletters++;
-      solutionArray[i] = null;
+    var solutionArray = solution.split('');
+    var guessArray = guess.split('');
+    var correctLetterLocations = 0;
+    var correctLetters = 0;
+    for (var i = 0; i<= solution.length-1; i++) {
+      if (solutionArray[i] === guessArray[i]) {
+        correctLetterLocations++ ;
+        solutionArray[i] = null;
```

Comment:
Great work on this app! Got the colors working on my end while running your app, so brownie points! As a concept question, why are we setting the checked index of solutionArray to null?